### PR TITLE
Enabled errorcheck linter and handled errors 🏃

### LINKED ISSUE #1541: cleanup: enable `errcheck` linter
After #1535, we updated the golangci-lint to the latest version and matched the upstream Cluster API format for the linter, it seems there were a bunch of linters missing.

- Head over to [.golanci-lint.yaml](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/.golangci.yml) in the root of this repository.
- Remove the linter from the `linters.disable` list.
- Run `make lint-full`.
- Fix the reported errors with the suggestions.
- Submit a PR with the changes.

Feel free to reach out at any time if you need guidance or help.

/kind cleanup
/help
/good-first-issue
/milestone Next
/priority backlog

### PR DESCRIPTION
**Pull request by teaguecole**
**What this PR does / why we need it**:
This PR enables the errorcheck linter. We need this because it follows best practices and makes for better Go code.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1541 

While the linter does not present any issues, I am not sure if I am handling every error the proper way.

### CODE REVIEW

File: controllers/awsmachine_controller_test.go [hunk 1]
```diff
@@ -298,7 +305,10 @@ var _ = Describe("AWSMachineReconciler", func() {
 
 				It("should not tag anything if there's not tags", func() {
 					ec2Svc.EXPECT().UpdateInstanceSecurityGroups(gomock.Any(), gomock.Any()).Times(0)
-					reconciler.reconcileNormal(context.Background(), ms, cs)
+					_, err := reconciler.reconcileNormal(context.Background(), ms, cs)
```
Comment (detiber) [inline controllers/awsmachine_controller_test.go]: nit: collapse this down into:
```go
if _, err := reconciler.reconcileNormal(context.Background(), ms, cs); err != nil {
```

File: pkg/cloud/services/ec2/securitygroups.go [hunk 2]
```diff
@@ -161,7 +161,10 @@ func (s *Service) deleteSecurityGroups() error {
 	}
 
 	for _, sg := range s.scope.SecurityGroups() {
-		s.deleteSecurityGroup(&sg, "managed")
+		err := s.deleteSecurityGroup(&sg, "managed")
```
Comment (detiber) [inline pkg/cloud/services/ec2/securitygroups.go]: nit: collapse this down into:
```go
if err := s.deleteSecurityGroup(&sg, "managed"); err != nil {
```

File: pkg/cloud/services/ec2/subnets_test.go [hunk 3]
```diff
@@ -34,7 +35,10 @@ import (
 )
 
 func init() {
-	clusterv1.AddToScheme(scheme.Scheme)
+	err := clusterv1.AddToScheme(scheme.Scheme)
```
Comment (detiber) [inline pkg/cloud/services/ec2/subnets_test.go]: nit: collapse this down into:
```go
if err := clusterv1.AddToScheme(scheme.Scheme); err != nil {
```

File: pkg/cloud/services/elb/loadbalancer.go [hunk 4]
```diff
@@ -132,7 +132,10 @@ func (s *Service) DeleteLoadbalancers() error {
 
 	for _, elb := range elbs {
 		s.scope.V(3).Info("deleting load balancer", "arn", elb)
-		s.deleteClassicELB(elb)
+		err = s.deleteClassicELB(elb)
```
Comment (detiber) [inline pkg/cloud/services/elb/loadbalancer.go]: nit: collapse this down into:
```go
if err := s.deleteClassicELB(elb); err != nil {
```

File: pkg/cloud/services/secretsmanager/cloudinit.go [hunk 5]
```diff
@@ -72,7 +72,10 @@ func GenerateCloudInitMIMEDocument(secretPrefix string, chunks int32, region str
 	}
 
 	var scriptBuf bytes.Buffer
-	secretFetchTemplate.Execute(&scriptBuf, scriptVariables)
+	err = secretFetchTemplate.Execute(&scriptBuf, scriptVariables)
```
Comment (detiber) [inline pkg/cloud/services/secretsmanager/cloudinit.go]: nit: collapse this down into:
```go
if err := secretFetchTemplate(&scriptBuf, scriptVariables); err != nil {
```

File: pkg/cloud/services/secretsmanager/helpers_test.go [hunk 6]
```diff
@@ -37,7 +37,10 @@ func TestSplitBytes(t *testing.T) {
 	t.Run("should call 1 time if it fits", func(t *testing.T) {
 		maxSize := 100
 		input := make([]byte, 100)
-		crand.Read(input)
+		_, err := crand.Read(input)
```
Comment (detiber) [inline pkg/cloud/services/secretsmanager/helpers_test.go]: nit: collapse this down into:
```go
if _, err := crand.Read(input); err != nil {
```

File: pkg/cloud/services/secretsmanager/helpers_test.go [hunk 7]
```diff
@@ -50,7 +53,10 @@ func TestSplitBytes(t *testing.T) {
 	t.Run("should properly split given random input and maxsize", func(t *testing.T) {
 		maxSize := 1 + rand.Intn(1024)
 		input := make([]byte, rand.Intn(24576))
-		crand.Read(input)
+		_, err := crand.Read(input)
```
Comment (detiber) [inline pkg/cloud/services/secretsmanager/helpers_test.go]: nit: collapse this down into:
```go
if _, err := crand.Read(input); err != nil {
```

File: pkg/internal/hash/base36.go [hunk 8]
```diff
@@ -32,7 +32,10 @@ func Base36TruncatedHash(str string, len int) (string, error) {
 		return "", errors.Wrap(err, "unable to create hash function")
 	}
 
-	hasher.Write([]byte(str))
+	_, err = hasher.Write([]byte(str))
```
Comment (detiber) [inline pkg/internal/hash/base36.go]: nit: collapse this down into:
```go
if _, err := hasher.Write([]byte(str)); err != nil {
```

File: cmd/release/main.go [hunk 9]
```diff
@@ -317,7 +317,10 @@ func (g git) getUserConfirmation(message string) {
 	reader := bufio.NewReader(os.Stdin)
 	g.logger.Info("Press return to confirm.")
 	// throwing away the error
-	reader.ReadString('\n')
+	_, err := reader.ReadString('\n')
```
[the remark this answers is not in the document]
Comment (teaguecole) [reply cmd/release/main.go]: file was removed in previous PR, had to rebase.

**Comment by teaguecole** [comment]
Some of your comments were folded in and so I didn't see them.  Now I do...

**Pull request merged**